eva_submission/eload_validation.py

```diff
@@ -80,6 +91,37 @@ def _validate_sample_names(self):
             })
         self.eload_cfg.set('validation', 'sample_check', 'pass', value=not overall_differences)
 
+    def detect_and_optionally_merge(self, merge_per_analysis):
+        """Detects merge type for each analysis, but performs merge only when merge_per_analysis is True."""
+        vcfs_by_analysis = self._get_valid_vcf_files_by_analysis()
+        vcfs_to_horizontal_merge = {}
+        vcfs_to_vertical_concat = {}
+        for analysis_alias, vcf_files in vcfs_by_analysis.items():
+            merge_type = detect_merge_type(vcf_files)
+            self.eload_cfg.set('validation', 'merge_type', analysis_alias, value=merge_type)
```

Comment:
Storing the `merge_type` is good but we should also provide it to the user in the report.